### DISCUSSION TITLE
[BUG] uploadListingVideo call Server error when trying to associate existing video

### DISCUSSION THREAD

**Discussion by valigara** [etsy/open-api#672] [Q&A] [answered]
Steps to reproduce:
1. Upload listing video
```Response
HTTP/1.1 201 Created
X-Etsy-Request-Uuid: EuplL2tWpBAg4gvnh_IKUw3Zls57

{
    "video_id": 723094608,
    "video_url": "https:\/\/v-cg.etsystatic.com\/video\/upload\/ac_none,du_15,q_auto:good\/USD%7Cen-US%7CUS_85f5f5fdf1e161f062cf59612fb2e088.mp4",
    "thumbnail_url": "https:\/\/v-cg.etsystatic.com\/video\/upload\/ar_1:1,c_fill,h_105,q_auto,w_105\/USD%7Cen-US%7CUS_85f5f5fdf1e161f062cf59612fb2e088.jpg",
    "height": 240,
    "width": 320,
    "video_state": "active"
}
```
2. Remove video from listing
Request:
```
DELETE /v3/application/shops/6897505/listings/1237742289/videos/723094608 HTTP/1.1
Host: openapi.etsy.com
Accept: application/json
x-api-key: ***
authorization: Bearer ***
```
Response:
```
HTTP/1.1 204 No Content
X-Etsy-Request-Uuid: EuMURSXA1d9hqkWNFFQB-GsJxI8b
```
3. Try to associate video to listing
Request:
```
POST /v3/application/shops/6897505/listings/1237742289/videos HTTP/1.1
Content-Type: application/x-www-form-urlencoded
Host: openapi.etsy.com
Accept: application/json
x-api-key: ***
authorization: Bearer ***

video_id=723094608
```
Response:
```
HTTP/1.1 500 Internal Server Error
Connection: keep-alive
Content-Length: 25
Server: Apache
X-CLOUD-TRACE-CONTEXT: 4564ffb30e21186ae0a26f1b105165b0/1157364580740230293;o=0
X-Etsy-Request-Uuid: Eu9D7nytk35Ve6bZAKZNlwgkB00c
X-Limit-Per-Second: 20
X-Remaining-This-Second: 19
X-Limit-Per-Day: 60000
X-Remaining-Today: 59728
Content-Type: application/json
Via: 1.1 google, 1.1 varnish
Accept-Ranges: bytes
Date: Tue, 22 Nov 2022 13:52:18 GMT
X-Served-By: cache-lcy-eglc8600048-LCY
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1669125138.415144,VS0,VE142

{"error":"Server Error"}
```

or in different way 
Request:
```
POST /v3/application/shops/6897505/listings/1237742289/videos HTTP/1.1
Content-Type: multipart/form-data; boundary=4255680b2752bb4b027e8dde688c053a207ef755
Host: openapi.etsy.com
Accept: application/json
x-api-key: ***
authorization: Bearer ***

--4255680b2752bb4b027e8dde688c053a207ef755
Content-Disposition: form-data; name="video_id"
Content-Length: 9

723094608
--4255680b2752bb4b027e8dde688c053a207ef755--
```
Response:
```
HTTP/1.1 500 Internal Server Error
Connection: keep-alive
Content-Length: 25
Server: Apache
X-CLOUD-TRACE-CONTEXT: ff18a1648388cd186edbbfd4a96527ac/2070692893553475659;o=0
X-Etsy-Request-Uuid: EuMUXYgD5p29UapTN8Mb9XVGLrdb
X-Limit-Per-Second: 20
X-Remaining-This-Second: 19
X-Limit-Per-Day: 60000
X-Remaining-Today: 59727
Content-Type: application/json
Via: 1.1 google, 1.1 varnish
Accept-Ranges: bytes
Date: Tue, 22 Nov 2022 13:58:00 GMT
X-Served-By: cache-mrs10533-MRS
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1669125480.490527,VS0,VE152

{"error":"Server Error"}
```

According to docs `uploadListingVideo`:
> Uploads a new video for a listing, or **associates an existing video with a specific listing**. You must either provide the **video_id** of an existing video, or the name and binary file data for a video to upload.

+ `deleteListingVideo` docs:
> Open API V3 endpoint to delete a listing video. **A copy of the video remains on our servers, and so a deleted video may be re-associated with the listing without re-uploading the original video**; see uploadListingVideo.

### Please advise what should be done to avoid `Server Error`?

**Comment by baahar** [comment]
I'm having the same issue. Uploading the same video as byte data anew causes no problem, but trying to associate a listing with an existing video via the video_id parameter returns a 500-Internal Server Error.

**Reply by chasemorell** [reply]
I am also having this **exact** same issue. I get a 500 Server Error.

**Reply by tmasopust** [reply]
I'm seeing the same issue. 

My assumption is that if you have the same video to apply to multiple listings, you want to share the resource vs. duplicating the same resource in storage. The API seems to support this notion by allowing you to associate a listing with a video_id. Perhaps this is not the case.

The 500 server error has no detail.

**Comment by Tony-Pappas** [comment]
This was reported on Nov 22, 2022 and they still haven't fixed it. What a total waste of resources - I have thousands of listings I have to upload this one video to. Guys, if you're going to put a bug on the backburner, at least indicate there's an issue in the docs so you're not wasting our time; I spent hours on this today.

**Comment by NickMihaiu91** [comment]
Still not fixed. This is outrageous. At least don't write in the docs this is possible and have us waste so much time figuring out what we did wrong when the actual API is broken.

**Comment by openapi-support** [comment, marked as answer]
Hi, we are so sorry that this issue has persisted for so long. A fix has been implemented as well as additional functionality to the `ListingVideo_Upload` endpoint. Please see our latest [release notes](https://github.com/etsy/open-api/releases) for the details. Thank you for bringing this to our attention and for your continued support of our API!

**Reply by valigara** [reply]
First of all thank you, but...
almost :)
According to out latest test when upload listing to video, delete it from video and assign back it returns proper api response,
but in Etsy ui video is still empty:
<img width="573" height="293" alt="image" src="https://github.com/user-attachments/assets/59e86242-816c-4c94-b0eb-f4ae9741c8f5" />

upload video file request
```
POST /v3/application/shops/6897505/listings/1417447458/videos HTTP/1.1
Expect: 100-Continue
Content-Type: multipart/form-data; boundary=d226b9d90cf8f9ffaed4216b57db00c70974760a
Host: openapi.etsy.com
Accept: application/json
x-api-key: ***
authorization: Bearer ***

```
response:
```
HTTP/1.1 201 Created
x-etsy-request-uuid: Eucr4nA09oYfjE4zB5CPWOHBXCf0

{"video_id":816129238,"video_url":"https:\/\/v.etsystatic.com\/video\/upload\/ac_none,du_15,q_auto:good\/USD%7Cen-US%7CUS_89edd21875d7f898933d888c591e4958.mp4","thumbnail_url":"https:\/\/v.etsystatic.com\/video\/upload\/ar_1:1,c_fill,h_105,q_auto,w_105\/USD%7Cen-US%7CUS_89edd21875d7f898933d888c591e4958.jpg","height":848,"width":848,"video_state":"active"}
```

delete video from listing request:
```
DELETE /v3/application/shops/6897505/listings/1417447458/videos/816129238 HTTP/1.1
Host: openapi.etsy.com
Accept: application/json
x-api-key: ***
authorization: Bearer ***

```
response:
```
HTTP/1.1 204 No Content
x-etsy-request-uuid: Euu2H_p_5TWqXNp9wiHnmGKcGP6d

```

assign video to same listing back using `816129238` video id returned from first call request:
```
POST /v3/application/shops/6897505/listings/1417447458/videos HTTP/1.1
Content-Type: multipart/form-data; boundary=8ac8ba97221776897f925a9d4222efcda0faa4be
Host: openapi.etsy.com
Accept: application/json
x-api-key: bz66***
authorization: Bearer 1916***

--8ac8ba97221776897f925a9d4222efcda0faa4be
Content-Disposition: form-data; name="video_id"
Content-Length: 9

816129238
--8ac8ba97221776897f925a9d4222efcda0faa4be--
```
response:
```
HTTP/1.1 201 Created
x-etsy-request-uuid: Eu9_8XcVxjm9w_P9VrpKdFd8kob1

{"video_id":816129238,"video_url":"https:\/\/v.etsystatic.com\/video\/upload\/ac_none,du_15,q_auto:good\/USD%7Cen-US%7CUS_89edd21875d7f898933d888c591e4958.mp4","thumbnail_url":"https:\/\/v.etsystatic.com\/video\/upload\/ar_1:1,c_fill,h_105,q_auto,w_105\/USD%7Cen-US%7CUS_89edd21875d7f898933d888c591e4958.jpg","height":848,"width":848,"video_state":"active"}
```

**Such bug is more dangerous, because despite success response from api in real video is not attached.**
We are hoping that addressing this bug will not take same amount of time as original one.

**Comment by openapi-support** [comment]
Thank you for pointing that out - this issue has been fixed.

**Reply by valigara** [reply]
Tested - confirmed - working as expected.
Thank you!
